src/Curiosity/Runtime.hs

```diff
@@ -350,22 +344,19 @@ createUser runtime (newId, User.Signup {..}) = do
 createUserFull
   :: Runtime -> User.UserProfile -> STM (Either User.UserErr User.UserId)
 createUserFull runtime newProfile = do
-  mprofile <- selectUserById runtime newProfileId
+  mprofile <- liftA2 (<|>)
+                     (selectUserById runtime newProfileId)
+                     (selectUserByUsername runtime userName)
   case mprofile of
-    Just profile -> existsErr
-    Nothing      -> createNew
+    Just _  -> existsErr
```

Comment:
I usually keep those unused variables, because the fix is to use them (i.e. parametrize the error message depending on the variable).